core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -50,6 +50,43 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   KOKKOS_IMPL_HIP_SAFE_CALL(
       hipGetDeviceProperties(&Impl::HIPInternal::m_deviceProp, hip_device_id));
   KOKKOS_IMPL_HIP_SAFE_CALL(hipSetDevice(hip_device_id));
+
+  // Check that we are running on the expected architecture
+  auto check_architecture = [](const std::string& gfx_name) {
+    std::string arch_name = Impl::HIPInternal::m_deviceProp.gcnArchName;
```

Comment:
I think these are good use cases for std::string_view (even though obviously it is not critical)